Misc dynamodb integration test improvements

mrk-its opened the pull request:
* add run_id column to test dataset (it allows for parallel test runs with proper row counting)
* general cleanup

mrk-its committed:
DynamoDBLogStore (#339) - initial Scala version + integration tests

* DynamoDBLogStore: initial version

* - Refactoring
- Refactor and reformat
- Add test cases
- Update after review
- Add DynamoDB tests
- Fix test name
- Adopt new tests
- Add docker support for tests

Ignore DynamoDB test because of CircleCI doesn't support testcontainers in docker

* bugfixes

* move to contribs

* add tests

* store temp files in .temp directory

* scala style, method rename

* revert .circleci changes, update dynamodb_logstore.py

* make AWS credentials configurable

* move example to contrib subdir

* refactoring

* add python integration test

* add --jars option

* change log levels

* change the test skipping logic and adding logging

* after-review changes

* store tablePath in db instead of _delta_log path

* add example invocation of integration test

* move FailingDynamoDBLogStore to test package

* refactor error injection

* changes after review

* store temp files inside of _delta_log directory

* linting

* fix docs

* remove unnecessary logic from getTablePath, fix tests

* minor fix

* move to storage-dynamodb

* storage-dynamodb project: add TODO, linting

Co-authored-by: Mateusz Sabat <mateusz.sabat@samba.tv>
Co-authored-by: Lukasz_P <lukasz.ponikiewski@samba.tv>
Co-authored-by: L-Po <ponikiewski.lukasz@gmail.com>

scottsand-db committed:
Fix python integration tests (#1009)

* fixed

* update example command; add TODOs to build.sbt

* fix order of repos argument in spark submit cmd

allisonport-db committed:
Refactor DynamoDB Logstore integration tests (#1010)

* initial refactoring

* save work

* clean up

* use environmental variable for logstore selection

* rename args

allisonport-db committed:
Merge remote-tracking branch 'delta-io/master' into dynamodb_logstore_scala_feature_branch

mrk-its committed:
Dynamodb logstore java (#1023)

* java refactor

* auto table creation

* auto-create dynamodb table

* minor changes:
 - create dynamodb client in constructor
 - do not apply extra filtering in listFrom
 - inline comments
 - add docs

scottsand-db committed:
Merge remote-tracking branch 'upstream/master' into dynamodb_logstore_java_feature_branch

mrk-its committed:
Misc dynamodb integration test improvements:
 - add run_id column
 - cleanup

scottsand-db committed:
Improvements to DynamoDBLogStore.java feature  #1 (#1039)

scottsand-db commented on storage-dynamodb/integration_tests/dynamodb_logstore.py:
```diff
@@ -90,24 +81,41 @@
+data = spark.createDataFrame([], SCHEMA)
+
+if table_overwrite:
+    data.write.format("delta").mode("overwrite").partitionBy("run_id", "id").save(delta_table_path)
```
should we also empty the DDB table?

mrk-its replied:
It is not necessary, if we want to start with no dynamodb entries for table we can create table in new location.

mrk-its replied:
But maybe we can simply remove this whole step? The intention was to create empty table before starting multi-thread writing, but maybe it is not necessary. I'll test it

mrk-its replied:
Ok, something interesting happened when I ran this test without this 'table creation' step:
```
delta.exceptions.ProtocolChangedException: The protocol version of the Delta table has been changed by a concurrent update. This happens when multiple writers are writing to an empty directory. Creating the table ahead of time will avoid this conflict. Please try the operation again.
Conflicting commit: {"timestamp":1648832395501,"operation":"WRITE","operationParameters":{"mode":Append,"partitionBy":["run_id","id"]},"isolationLevel":"Serializable","isBlindAppend":true,"operationMetrics":{"numFiles":"1","numOutputRows":"1","numOutputBytes":"449"},"engineInfo":"Apache-Spark/3.2.0 Delta-Lake/1.2.0-SNAPSHOT","txnId":"bcd0f41e-d9ff-4455-ad02-e18d57c150d2"}
```
The interesting thing it failed with above message after inserting ~50 row to this table (I would expect quick fail with no rows inserted at all). Probably single thread failed after some timeout while other threads were inserting data.

mrk-its replied:
Ok - I simplified whole thing by removing this conditional overwrite. Instead table is created if necessary by appending  empty data unconditionally.

scottsand-db commented:
@mrk-its if this PR is ready for review  can you remove the Draft status?

mrk-its commented:
> @mrk-its if this PR is ready for review can you remove the Draft status?

done

scottsand-db committed:
Improvements to DynamoDBLogStore.java feature #2 (#1047)

* minor cleanup

* add print statement to dynamodb_logstore.py

* add comment to build.sbt on how to run FailingDynamoDBLogStore

* rename api conf keys

* Update dynamodb_logstore.py

scottsand-db committed:
update configuration prefix

scottsand-db commented:
will review once conflicts are resolved

scottsand-db committed:
Misc fix

- update build.sbt to remove aws sdk from contribs
- update BaseExternalLogStore to use FileNameUtils

scottsand-db commented:
@mrk-its can you please just rebase the commit https://github.com/delta-io/delta/pull/1046/commits/840710679d1ac080e954a99f2469d34709faeffa on top of master?

foss-contributor commented:
@mrk-its, thank you for your contribution! This pull request has been inactive for over 18 months and will be closed in one week. To keep it open, please leave a comment on this pull request or contact us in the `#contributors` channel on [Delta Lake Slack](https://go.delta.io/slack). We appreciate your involvement in Delta Lake!
